src/jit/compiler.h

```diff
@@ -857,50 +857,206 @@ const unsigned int   MAX_INL_ARGS =      10;     // does not include obj pointer
 const unsigned int   MAX_INL_LCLS =      8;
 #endif // LEGACY_BACKEND
 
+// InlineResult describes the various states the jit goes through when
+// evaluating an inline candidate. It is distinct from CorInfoInline
+// because it must capture interal states that don't get reported back
+// to the runtime.
+
+enum class InlineResult {
```

Comment:
Would it make sense to give this a different anme, such as InlineDecision, to distinguish it from JitInlineResult more clearly?